packages/react-router-dom/__tests__/special-characters-test.tsx

```diff
@@ -72,70 +81,123 @@ let specialChars = [
 
   // These chars get encoded in all portions of the URL
   {
     char: "🤯",
     pathChar: "%F0%9F%A4%AF",
     searchChar: "%F0%9F%A4%AF",
     hashChar: "%F0%9F%A4%AF",
+    decodedChar: "🤯",
   },
   {
     char: "✅",
     pathChar: "%E2%9C%85",
     searchChar: "%E2%9C%85",
     hashChar: "%E2%9C%85",
+    decodedChar: "✅",
   },
   {
     char: "🔥",
     pathChar: "%F0%9F%94%A5",
     searchChar: "%F0%9F%94%A5",
     hashChar: "%F0%9F%94%A5",
+    decodedChar: "🔥",
+  },
+  {
+    char: "ä",
+    pathChar: "%C3%A4",
+    searchChar: "%C3%A4",
+    hashChar: "%C3%A4",
+    decodedChar: "ä",
+  },
+  {
+    char: "Ä",
+    pathChar: "%C3%84",
+    searchChar: "%C3%84",
+    hashChar: "%C3%84",
+    decodedChar: "Ä",
+  },
+  {
+    char: "ø",
+    pathChar: "%C3%B8",
+    searchChar: "%C3%B8",
+    hashChar: "%C3%B8",
+    decodedChar: "ø",
   },
-  { char: "ä", pathChar: "%C3%A4", searchChar: "%C3%A4", hashChar: "%C3%A4" },
-  { char: "Ä", pathChar: "%C3%84", searchChar: "%C3%84", hashChar: "%C3%84" },
-  { char: "ø", pathChar: "%C3%B8", searchChar: "%C3%B8", hashChar: "%C3%B8" },
   {
     char: "山",
     pathChar: "%E5%B1%B1",
     searchChar: "%E5%B1%B1",
     hashChar: "%E5%B1%B1",
+    decodedChar: "山",
   },
   {
     char: "人",
     pathChar: "%E4%BA%BA",
     searchChar: "%E4%BA%BA",
     hashChar: "%E4%BA%BA",
+    decodedChar: "人",
   },
   {
     char: "口",
     pathChar: "%E5%8F%A3",
     searchChar: "%E5%8F%A3",
     hashChar: "%E5%8F%A3",
+    decodedChar: "口",
   },
   {
     char: "刀",
     pathChar: "%E5%88%80",
     searchChar: "%E5%88%80",
     hashChar: "%E5%88%80",
+    decodedChar: "刀",
   },
   {
     char: "木",
     pathChar: "%E6%9C%A8",
     searchChar: "%E6%9C%A8",
     hashChar: "%E6%9C%A8",
+    decodedChar: "木",
   },
 
   // Add a few multi-char space use cases for good measure
-  { char: "a b", pathChar: "a%20b", searchChar: "a%20b", hashChar: "a%20b" },
-  { char: "a+b", pathChar: "a+b", searchChar: "a+b", hashChar: "a+b" },
+  { char: "a b", pathChar: "a%20b", searchChar: "a%20b", hashChar: "a%20b", decodedChar: "a b" },
+  { char: "a+b", pathChar: "a+b", searchChar: "a+b", hashChar: "a+b", decodedChar: "a+b" },
+
+  // Edge case scenarios where the incoming `char` (or string) is pre-encoded
+  // because it contains special characters such as `&`, `%`, or `#`.  For these
+  // we provide a `decodedChar` so we can assert the param value gets decoded
+  // properly and so we can ensure we can match these decoded values in static
+  // paths
+  {
+    char: "a%25b",
+    pathChar: "a%25b",
+    searchChar: "a%25b",
+    hashChar: "a%25b",
+    decodedChar: "a%b",
+  },
+  {
+    char: "a%23b%25c",
+    pathChar: "a%23b%25c",
+    searchChar: "a%23b%25c",
+    hashChar: "a%23b%25c",
+    decodedChar: "a#b%c",
+  },
+  {
+    char: "a%26b%25c",
+    pathChar: "a%26b%25c",
+    searchChar: "a%26b%25c",
+    hashChar: "a%26b%25c",
+    decodedChar: "a&b%c",
+  },
```

Comment:
These are the net new tests for the bugs being reported, if you pre-encode because that portion of the URL has URL-sensitive characters - you weren't previously getting back the proper decoded value.  These are the only tests right now that have a different `char` versus `decodedChar`